source/server/init_manager_impl.cc

```diff
@@ -28,7 +28,7 @@ void InitManagerImpl::initialize(std::function<void()> callback) {
 void InitManagerImpl::initializeTarget(Init::Target& target) {
   target.initialize([this, &target]() -> void {
     ASSERT(std::find(targets_.begin(), targets_.end(), &target) != targets_.end());
-    targets_.remove(&target);
+    targets_.erase(&target);
```

Comment:
One thing to keep in mind is that this method is called from `initialize`, which is iterating over the container. If these callbacks fire immediately, i.e. are not deferred, then you need to make sure the removal process doesn't invalidate the iterator. This is explicitly support for `std::list`, see https://en.cppreference.com/w/cpp/container/list, but I don't know if it works for unordered set.

Reply:
We don't actually rely on that guarantee, as far as I can tell... See the comment in `initialize` and the careful attention that's paid to storing the target, then incrementing the iterator, and then calling `initializeTarget` to avoid the iterator invalidation issue.

Reply:
That's true, so, yeah, this is safe.